### PR TITLE
[DI] Dont call get() when not required - µ-optim

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

(could be applied on 3.2, please advise)
Helps a bit on https://rawgit.com/kocsismate/php-di-container-benchmarks/master/var/benchmark.html

note: if we want more performance, we should consider deprecating case insensitivity of service identifiers.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1505,7 +1505,7 @@ private function getServiceCall($id, Reference $reference = null)
         if ($this->container->hasDefinition($id) && !$this->container->getDefinition($id)->isPublic()) {
             // The following is PHP 5.5 syntax for what could be written as "(\$this->services['$id'] ?? \$this->{$this->generateMethodName($id)}())" on PHP>=7.0
```
Comment (lyrixx) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: This comment could be removed?!

**Review by lyrixx — COMMENTED** [review]
:+1: The new code is so much simpler ;) Thanks.

**Comment by nicolas-grekas** [comment]
@lyrixx unfortunately I had to revert to the previous syntax to make it work on PHP 5

**Comment by fabpot** [comment]
👍

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by theofidry** [comment]
@nicolas-grekas did you manage to get any insight regarding the perf improvement? I didn't get any significant diff with blackfire (granted phpbench would be more suitable for that work) with that change.

**Comment by nicolas-grekas** [comment]
Yes, see https://github.com/kocsismate/php-di-container-benchmarks/pull/8

**Comment by kocsismate** [comment]
hey @nicolas-grekas and @theofidry 

I've just discovered this PR and I also became curious about the performance implications of this change. Fortuantely, I managed to make my test suite much more reliable in the past weeks (mainly by running it on my machine with PHP-FPM with Opcache enabled).

Test Suite 1-2 and 5-6 are relevant for the change you made. The difference between them is that Test Suite 5 and 6 don't measure autoload and bootstrap times at all - so containers have already become "warm" before measuring takes place. Unfortunately, I saw a modest performance drop in the first two test suites, and no change in the last two:

- Results for Symfony DI 3.2: https://rawgit.com/kocsismate/php-di-container-benchmarks/d3b92f337a3e38a4066f892ef77d2445fffc24c3/var/benchmark.html
- Results for Symfony DI 3.3: https://rawgit.com/kocsismate/php-di-container-benchmarks/a7a36d38bf4d9295fef14c6f391785db6d800945/var/benchmark.html

I ran the tests multiple times but the outcome didn't change significantly. That's all what I could find out so far. :)

**Pull request merged**